src/ripple/app/paths/PathRequest.cpp

```diff
@@ -175,70 +166,69 @@ void PathRequest::updateComplete ()
-            jvStatus = rpcError (rpcSRC_ACT_NOT_FOUND);
-        }
+        // if send max specified, dst amt must be -1
+        jvStatus = rpcError(rpcDST_AMT_MALFORMED);
+        return false;
     }
 
-    if (bValid)
+    if (! crCache->getLedger()->exists(
+        keylet::account(*raSrcAccount)))
     {
-        auto const sleDest = crCache->getLedger()->read(
-            keylet::account(*raDstAccount));
+        // source account does not exist
+        jvStatus = rpcError (rpcSRC_ACT_NOT_FOUND);
+        return false;
+    }
 
-        Json::Value& jvDestCur =
-                (jvStatus[jss::destination_currencies] = Json::arrayValue);
+    auto const& lrLedger = crCache->getLedger();
+    auto const sleDest = crCache->getLedger()->read(
+        keylet::account(*raDstAccount));
 
-        if (!sleDest)
-        {
-            // no destination account
-            jvDestCur.append (Json::Value ("XRP"));
+    Json::Value& jvDestCur =
+        (jvStatus[jss::destination_currencies] = Json::arrayValue);
 
-            if (!saDstAmount.native ())
-            {
-                // only XRP can be send to a non-existent account
-                bValid = false;
-                jvStatus = rpcError (rpcACT_NOT_FOUND);
-            }
-            else if (saDstAmount < STAmount (lrLedger->fees().accountReserve (0)))
-            {
-                // payment must meet reserve
-                bValid = false;
-                jvStatus = rpcError (rpcDST_AMT_MALFORMED);
-            }
+    if (! sleDest)
+    {
+        // no destination account
+        jvDestCur.append (Json::Value ("XRP"));
+        if (! saDstAmount.native ())
+        {
+            // only XRP can be send to a non-existent account
+            jvStatus = rpcError (rpcACT_NOT_FOUND);
+            return false;
         }
-        else
+
+        if (! convert_all_ && saDstAmount <
+            STAmount (lrLedger->fees().accountReserve (0)))
         {
-            bool const disallowXRP (
-                sleDest->getFlags() & lsfDisallowXRP);
+            // payment must meet reserve
+            jvStatus = rpcError (rpcDST_AMT_MALFORMED);
+            return false;
+        }
+    }
+    else
+    {
+        bool const disallowXRP (
+            sleDest->getFlags() & lsfDisallowXRP);
 
-            auto usDestCurrID = accountDestCurrencies (
-                    *raDstAccount, crCache, !disallowXRP);
+        auto usDestCurrID = accountDestCurrencies (
+                *raDstAccount, crCache, ! disallowXRP);
 
-            for (auto const& currency : usDestCurrID)
-                jvDestCur.append (to_string (currency));
+        for (auto const& currency : usDestCurrID)
+            jvDestCur.append (to_string (currency));
 
-            jvStatus["destination_tag"] =
-                    (sleDest->getFlags () & lsfRequireDestTag)
-                    != 0;
-        }
+        jvStatus["destination_tag"] =
+                (sleDest->getFlags () & lsfRequireDestTag)
```

Comment:
Perhaps `bool(sleDest->getFlags () & lsfRequireDestTag);` is clearer?  Perhaps not?